### PR TITLE
feat: added a "testing" section to PR template

### PR DESCRIPTION
**Pull request by uostimb**
RFC

I discussed this with Graham last week and he agreed it could be a useful addition and was worthy of wider consideration, but it does not quite warrant a town-hall discussion in the Engineering Strategy meeting. 

## what.
* Add a new "testing" section to the default PR template.

## why.
* Prompt authors to clearly describe how their changes were tested and validated.
* Help PR reviewers identify potential gaps in testing and edge cases the author may have missed..
* Will encourage more thoughtful test coverage.
* Reduce the risk of problems making it into production, helping to prevent painful rollbacks (especially in features that include migrations) and improve confidence in production deployments.

## dependencies.
* None

## references.
* None

## platform diagrams.
* None

### CODE REVIEW

**Review by szbartnik — APPROVED** [review]
👏🏻

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,32 +1,64 @@
...
+-->
+
+- [ ] Unit tests added/updated (where needed)
+- [ ] Integration tests added/updated (where needed)
+- [ ] Automated QA test case added/updated (where needed)
+- [ ] Manual testing performed (where needed)
+
+**Notes / Evidence (optional):**
+
+<!-- Add reasoning if no tests were needed, or share logs/screenshots if manual testing was done. -->
```
Comment (uostimb) [inline .github/PULL_REQUEST_TEMPLATE.md:31]: I like this prompt to include logs/screenshots! 

I just wonder if (especially in the long term) it would end up being shortcutted by just pasting one or two token screenshots showing their feature in action on the happy path, rather than describing/demonstrating the testing steps they performed to give reviewers a chance to spot gaps and/or edge cases that were neglected, or even to follow-along-at-home and repeat the testing themselves if they so desire, which was more or less the entire reason for this PR.

I'm open to suggestion and negotiation, but how about:
```suggestion
<!-- If testing was not required, provide the reason. If manual testing was conducted, include logs/screenshots along with a description of the testing steps performed. -->
```

**Review by szbartnik — APPROVED** [review]
Looks lovely!
